platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1375,31 +1375,33 @@ void startUrlRequest(@NonNull final String url, final long requestHandle) {
             return;
         }
 
-        final Callback callback = new Callback() {
+        final HttpHandler.Callback callback = new HttpHandler.Callback() {
             @Override
-            public void onFailure(final Call call, final IOException e) {
+            public void onFailure(IOException e) {
                 nativeOnUrlComplete(mapPointer, requestHandle, null, e.getMessage());
             }
 
             @Override
-            public void onResponse(final Call call, final Response response) throws IOException {
-                if (!response.isSuccessful()) {
-                    nativeOnUrlComplete(mapPointer, requestHandle, null, response.message());
-                    throw new IOException("Unexpected response code: " + response + " for URL: " + url);
-                }
-                final ResponseBody body = response.body();
-                if (body == null) {
-                    nativeOnUrlComplete(mapPointer, requestHandle, null, response.message());
-                    throw new IOException("Unexpected null body for URL: " + url);
+            public void onResponse(final int code, final byte[] rawDataBytes, final Map<String, String> headers) {
+                // TODO: Use of returned error code and headers for better network response logging/retries, etc
+                if (code >= 200 && code < 300) {
+                    nativeOnUrlComplete(mapPointer, requestHandle, rawDataBytes, null);
                 }
                 else {
-                    final byte[] bytes = body.bytes();
-                    nativeOnUrlComplete(mapPointer, requestHandle, bytes, null);
+                    nativeOnUrlComplete(mapPointer, requestHandle, null, "Unexpected response code: " + code + " for URL: " + url);
```

Comment:
Why not pass in the body data here too?

Reply:
In the core code, if the `UrlResponse` has an error string, we never check the validity of the data bytes. Not sure we should pass in any data if there is an error associated the response.

Reply:
Fair enough - saves us a bit of work by not copying anything into a native vector.